WalletWasabi.Gui/Controls/TogglePasswordBox.cs

```diff
@@ -0,0 +1,61 @@
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
+using Avalonia.Data;
+using Avalonia.Styling;
+using System;
+using ReactiveUI;
+using System.Reactive.Linq;
+
+namespace WalletWasabi.Gui.Controls
+{
+	public class TogglePasswordBox : ExtendedTextBox, IStyleable
+	{
+		Type IStyleable.StyleKey => typeof(TogglePasswordBox);
+
+		public static readonly StyledProperty<bool> IsPasswordVisibleProperty =
+			AvaloniaProperty.Register<TogglePasswordBox, bool>(nameof(IsPasswordVisible), defaultBindingMode: BindingMode.TwoWay);
+
+		public bool IsPasswordVisible
+		{
+			get => GetValue(IsPasswordVisibleProperty);
+			set => SetValue(IsPasswordVisibleProperty, value);
+		}
+
+		public TogglePasswordBox()
+		{
+			UseFloatingWatermark = true;
+			Watermark = "Password";
+
+			this.GetObservable(IsPasswordVisibleProperty)
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(x =>
+			{
+				IsPasswordVisible = x;
+			});
+
+			this.WhenAnyValue(x => x.IsPasswordVisible)
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(x =>
+			{
+				PasswordChar = x ? '\0' : '*';
```

Comment:
@molnard can you try replacing the asterisks with `\u2022` ? it's a password black dot and i think it's a bit more familiar looking character (needs to be tested on all 3 platforms if they don't look like a square box).
![image](https://user-images.githubusercontent.com/16554748/64022293-73b83480-cb68-11e9-9583-773bb700ff21.png)

Reply:
> The warning message is clipping out when there isn't enough space. Per our chat, adding a tooltip that replicates the warning message would be the best approach right now.
> 
> ![image](https://user-images.githubusercontent.com/16554748/64021679-1a9bd100-cb67-11e9-928a-6da84e50d397.png)

Spent an hour with this. I added a `ToolTip.Tip` here:
https://github.com/molnard/WalletWasabi/blob/b39250665b88d6013e49540c0fed4133ba3486f2/WalletWasabi.Gui/Styles/Styles.xaml#L143
But how can I extract the proper message?

Reply:
I think you can add the ff to the Error ContentControl:
```
ToolTip.Tip="{Binding (DataValidationErrors.Errors)[0].Message}" 
```
Rationale: DataValidationErrors.Errors returns a `List<Exception>` and Bindings can crawl that for us via the indexer `[0]`, then accessing the message via `.Message`.

Reply:
Need to modify the template to get the error message outside the box!

We could show a red exclamation icon and tooltip with the message.

Reply:
> @molnard can you try replacing the asterisk with `\u2022` ? it's a password black dot and i think it's a bit more familiar looking character (needs to be tested on all 3 platforms if they don't look like a square box).
> ![image](https://user-images.githubusercontent.com/16554748/64022293-73b83480-cb68-11e9-9583-773bb700ff21.png)

Tested on:

Win 10, Win7, Osx, Ubuntu. Looks fine. Only thing I am not sure what happens on some other rare linux distros. Anyway the cross-platform compatibility of this is good so I will use it.

Reply:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Default/TextBox.xaml#L33

https://github.com/zkSNACKs/WalletWasabi/blob/2214cb82d41ca7ec280f90c8d7c2228d057ebd4f/WalletWasabi.Gui/Styles/Styles.xaml#L134

here is how you change the way validation errors are presented.